Common/Orders/Fills/EquityFillModel.cs

```diff
@@ -590,59 +590,79 @@ public virtual OrderEvent MarketOnCloseFill(Security asset, MarketOnCloseOrder o
 
             var subscribedTypes = GetSubscribedTypes(asset);
 
-            // The official close is recorded after the market is closed
             if (subscribedTypes.Contains(typeof(Tick)))
             {
-                var primaryExchange = (byte)((Equity)asset).PrimaryExchange;
-                var officialClose = (uint)(TradeConditionFlags.Regular | TradeConditionFlags.OfficialClose);
-                var closingPrints = (uint)(TradeConditionFlags.Regular | TradeConditionFlags.ClosingPrints);
-                var ticks = asset.Cache.GetAll<Tick>().OrderBy(x => x.EndTime).ToList();
+                var ticks = asset.Cache.GetAll<Tick>()
+                    .Where(x => x.TickType == TickType.Trade && x.Price > 0)
+                    .OrderBy(x => x.EndTime).ToList();
 
-                // Get the last valid (non-zero) tick of trade type from an close market
-                var trade = ticks
-                    .Where(x => !string.IsNullOrWhiteSpace(x.SaleCondition))
-                    .LastOrDefault(x =>
-                        x.TickType == TickType.Trade && x.Price > 0 && x.ExchangeCode == primaryExchange &&
-                        (x.ParsedSaleCondition == officialClose || x.ParsedSaleCondition == closingPrints));
+                var trade = ticks.LastOrDefault();
 
-                // If there is no tick with the desired information, we wait one minute
-                if (trade == null)
+                // The official close is recorded after the market is closed,
+                // We can only use this information if extended market hours are included
+                if (Parameters.ConfigProvider.GetSubscriptionDataConfigs(asset.Symbol).IsExtendedMarketHours())
                 {
-                    trade = ticks.LastOrDefault(x => x.Price > 0);
-                    if (trade == null || (trade.EndTime - nextMarketClose).TotalMinutes < 1)
+                    var primaryExchange = (byte) ((Equity) asset).PrimaryExchange;
+                    var officialClose = (uint) (TradeConditionFlags.Regular | TradeConditionFlags.OfficialClose);
+                    var closingPrints = (uint) (TradeConditionFlags.Regular | TradeConditionFlags.ClosingPrints);
```

Comment:
Think it should still _try_ to get the official closing trade even if `IsExtendedMarketHours` is false, the difference would be in one case it can wait a bit in the other it wouldn't? Thinking if the implementation of `MarketOnOpenFill` should also have a timeout (probably the same one) for waiting for the official open tick similar case as this